transcom/mymove
# B 17906 int Prevent Customer Request after Authorized End Date

## [B-17906](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A867878)

## Summary
Follow on from B-17854.
This story prevents Prime from updating the dates for Authorized End Date if the new calculation would be after the current Authorized End Date. Customer dates should not extend the Authorized End Date beyond its current date. This avoids an automatic SIT Extension. 

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Functionality Test

1. Create a new HHG shipment in the Customer app and submit it
2. Login as a Service Counselor and approve the shipment
3. Login as a TOO and accept the shipment
4. Login as a Prime into the Prime Simulator
5. Select the newly created move
6. Update the shipment and make sure you enter values for the estimated and actual weights, this is needed to calculate the required delivery date. Note the weight entered.
7. Click Add Service Item
8. Add a new Origin SIT item, enter today's date for SIT Entry Date
9. Login as a TOO access the move and go to Move Task Order tab.
10. Accept all of the SIT Service Items
11. Login as a Prime to the Prime Simulator
12. Select the Move
13. Click on Edit for DOASIT
14. Enter SIT Departure Date, SIT Customer Contact Date make today and Customer Requested Delivery a few days after today
15. Click Save
16. Go back in and edit the dates
17. For all SIT dates enter dates that are about a month or more in the future to try and extend the SIT Authorized End Date.
18. Click Save
19. You should see an error "dates entered cannot extend authorized end date beyond its current date"

Authorized End Date for Origin SIT: sitAuthorizedEndDate should be GracePeriodDays days after sitCustomerContacted or the sitDepartureDate whichever is earlier.
![image](https://github.com/transcom/mymove/assets/149000222/5e7c7d42-198f-4b2f-9c81-2db985c6c380)

### Backend
